src/AbstractArray.php

```diff
@@ -180,7 +180,7 @@ public function __construct(array $elements = [])
      *
      * @return static The instance array with combined elements
      */
-    abstract public function combineTo(array $array);
+    abstract public function useAndCombineTo(array $array);
```

Comment:
This comment is for all changed method names If you give 2 verbs to a method then it may sound as if the method is doing 2 actions ... which is meeh IMHO. `replaceKeysWith` and `replaceValuesWith` that were proposed as possible alternative are better suited IMHO as they do convey what the method will do.
My 2 cents